exporter/opentelemetry-exporter-prometheus/src/opentelemetry/exporter/prometheus/__init__.py

```diff
@@ -112,8 +116,16 @@ class PrometheusMetricReader(MetricReader):
             the metric belongs to.
     """
 
-    def __init__(self, prefix: str = "") -> None:
-        super().__init__()
+    def __init__(
+        self,
+        preferred_temporality: Dict[type, AggregationTemporality] = None,
+        preferred_aggregation: Dict[type, Aggregation] = None,
+        prefix: str = "",
+    ) -> None:
+        super().__init__(
+            preferred_temporality=preferred_temporality,
```

Comment:
I might've missed the discussion for this but how does adding preferred temporality configuration solve the "previuous point" issue?

Reply:
@ocelotl This shouldn't be configurable. It should always be set to cumulative for the prometheus exporter.

Reply:
I think we have to define better the behavior that we expect from the Prometheus metric reader. I understand that we see the Prometheus metric reader as being "always cumulative" but this is not accurate. There are 2 points here:

1. In our implementation, temporality is not a concept associated with metric readers but with instrument types.
2. Metric readers inherit from a parent class that receives the dictionary to configure the association between instrument types and metric readers, and the child classes (like the Prometheus metric reader) must respect this so that the interface for instantiation of a metric reader is consistent.

So, how do we want to implement this "restriction" where a Prometheus metric reader must "be cumulative"? How about we allow the dictionary to be passed but we raise an exception if a instrument class is to be configured as something else than cumulative?

Reply:
We could create a dict with InstrumentType and the temporality set to CUMULATIVE for all of them.

Reply:
I am fine with your proposed solution as well.

Reply:
@aabmass Do you have thoughts on this?

Reply:
@srikanthccv PTAL, I added a fix that uses a default cumulative-only dictionary and an exception that is raised if another temporality is found.